src/main/webapp/js/datahandlers/collegepagedatahandler.js

```diff
@@ -39,32 +41,38 @@ const convertFromJsonToTemplate = async (json) => {
     netCostForFifthQuintile: json["netCostForFifthQuintile"],
     cumulativeMedianDebt: json["cumulativeMedianDebt"]
   };
-  return data;
 };
 
 /**
  * Fetch request to the data servlet and return the JSON response.
- * @return - The JSON response.
+ * @public
+ * @return {*} - The JSON response.
  */
 const loadCollegeJson = async () => {
-  try{
     const response = await fetch(ENDPOINT);
-    let json = await response.json();
-    return json;
-  } catch(err) {
-    alert('Failed to retrieve data from a single college.');
-    console.log(err);
-  }
+    if(response.ok) {
+      return await response.json();
+    } else {
+      throw new Error(`${response.statusText}. Status: ${response.status}`);
+    }
 };
 
 /**
  * Render the soy template's html by attaching it to the desired DOM element.
- * @param {*} element - The DOM element where the template should be rendered to.
+ * @public
+ * @param {Element} element - The DOM element where the template should be rendered to.
  */
 const loadCollegeData = async (element) => {
-  let json = await loadCollegeJson();
-  let data = await convertFromJsonToTemplate(json);
-  console.log(data);
+  let json = undefined;
+
+  try {
+    json = await loadCollegeJson();
+  } catch(err) {
+    alert(err);
+    console.log(`Failed to load college data: ${err}`);
```

Comment:
Handle this error without a console.log.

Reply:
Hah. I'm fine with not printing the error message to the console, but it will make your life easier with debugging. Pretty sure throwing will output the error in the console anyway.